addons/mail/models/discuss/discuss_channel_category.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class ChannelCategory(models.Model):
+    _name = "discuss.channel.category"
+    _description = "Channel Category"
+
+    name = fields.Char('Channel category name', required=True)
+    channel_ids = fields.One2many('discuss.channel', 'channel_category_id', string='Channels')
+
+    @api.model
+    def get_categories(self):
+        categories = self.env['discuss.channel.category'].search([])
+        return Store(categories).get_result()
```

Comment:
I'd prefer if we have a route rather than a public model method (more flexible, we can easily add guest if necessary, etc).